### PR TITLE
LPS-173080 OldValues: Unit tests and Integration tests with Actions

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-173080

### CODE REVIEW

**Comment by naraandrade** [comment]
ci:test:object

**Comment by naraandrade** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
Depends on: https://github.com/liferay-objects/liferay-portal/pull/1686 and https://github.com/liferay-objects/liferay-portal/pull/1688

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java [hunk 1]
```diff
@@ -801,6 +828,28 @@ private void _assertWebhookObjectAction(
...
+					QueryUtil.ALL_POS)) {
+
+			if (StringUtil.equals(
+					MapUtil.getString(objectEntry.getValues(), "delta"),
+					"successful")) {
+
+				_objectEntryLocalService.deleteObjectEntry(
+					objectEntry.getObjectEntryId());
+
+				return true;
```
[the remark this answers is not in the document]
Comment (p-albuquerque) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:857]: I will take a look

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java [hunk 2]
```diff
@@ -885,6 +934,281 @@ private void _publishCustomObjectDefinition() throws Exception {
...
+		throws Exception {
+
+		return _objectActionLocalService.updateObjectAction(
+			objectAction.getExternalReferenceCode(),
+			objectAction.getObjectActionId(), true, conditionExpression,
+			objectAction.getDescription(), objectAction.getErrorMessageMap(),
+			objectAction.getLabelMap(), objectAction.getName(),
+			objectAction.getObjectActionExecutorKey(),
+			objectAction.getObjectActionTriggerKey(),
+			objectAction.getParametersUnicodeProperties());
```
[the remark this answers is not in the document]
Comment (p-albuquerque) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:1219]: When I try it. I got this:

> OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.object.model.impl.ObjectActionImpl#45143]

[the remark this answers is not in the document]
Comment (p-albuquerque) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:1219]: Also, I've tried: 
`		objectAction.setConditionExpression("(oldValue(\\\"charlie\\\") == 0)");

		objectAction = _objectActionLocalService.updateObjectAction(objectAction);`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java [hunk 3]
```diff
@@ -885,6 +934,281 @@ private void _publishCustomObjectDefinition() throws Exception {
...
+
+		for (ObjectEntry objectEntry : objectEntries) {
+			_objectEntryLocalService.deleteObjectEntry(
+				objectEntry.getObjectEntryId());
+		}
+
+		_objectDefinitionLocalService.deleteObjectDefinition(
+			objectDefinition.getObjectDefinitionId());
+		_listTypeDefinitionLocalService.deleteListTypeDefinition(
+			listTypeDefinition.getListTypeDefinitionId());
```
[the remark this answers is not in the document]
Comment (p-albuquerque) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:1183]: The order matters here, I will add a blank line

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Pull request closed without merging**